cassandra/components.py

```diff
@@ -763,14 +762,14 @@ def run_component(self):
             return rc
         else:
             stderr.write(
-                '[HistoricalHydroComponent]: Some output files were not created.  Check logfile (%s) for details.\n' % logfile)
+                f'[HistoricalHydroComponent]: Some output files were not created.  Check logfile ({logfile}) for details.\n')
             return 1            # nonzero indicates failure
 
 # This is how you run the disaggregation code
 # matlab -nodisplay -nosplash -nodesktop -r "run_disaggregation('<runoff-file>', '<chflow-file>', '<gcam-filestem>');exit" >& <logfile> < /dev/null
 
 
-class WaterDisaggregationComponent(GcamComponentBase):
+class WaterDisaggregationComponent(ComponentBase):
```

Comment:
Just FYI, these components (i.e., the old water code) are all going to get replaced pretty soon here, so don't spend too much time cleaning them up, apart from whatever is required to avoid parse errors in python 3, of course.